test/metadata_test.go

```diff
@@ -113,7 +114,7 @@ func (s) TestInvalidMetadata(t *testing.T) {
 		if err := stream.Send(&testpb.StreamingOutputCallRequest{}); err != nil {
 			t.Errorf("call ss.Client stream Send(nil) will success but got err :%v", err)
 		}
-		if _, err := stream.Recv(); !reflect.DeepEqual(test.recv, err) {
+		if _, err := stream.Recv(); status.Code(err) != status.Code(test.recv) || !strings.Contains(err.Error(), test.recv.Error()) {
```

Comment:
Should we switch the `reflect.DeepEqual()` on line 100 to something similar to here?

Reply:
Those errors are produced by our metadata library, so should be reliable.

Reply:
What I meant was to get rid of the `DeepEqual()` and instead rely on direct comparison of the status code and the error string.

Reply:
The problem is not the `DeepEqual` but the fact that we're testing for an error string that comes out of the standard library.  I think the `DeepEqual` is fine and simpler than checking the components separately.